### ISSUE TITLE
cinnamon: menu size permanently reduced after showing OSK

### ISSUE THREAD

**Issue by haggen88** [linuxmint/mint21.3-beta#23]
Unlike apps like e.g. Nemo, the menu does not return to its previous size.

[cinnamon menu.webm](https://github.com/linuxmint/mint21.3-beta/assets/68134588/13df5c4f-2c54-421e-8383-6ff3a8210530)

steps to follow:
1 open the virtual keyboard
2 open the menu and navigate through it
3 close the virtual keyboard
4 reopen the menu

**Comment by MenthaSuaveolens** [comment]
I can reproduce this on LM 21.2. Open Virtual keyboard, then open Cinnamon menu. At first it overlaps the virtual keyboard, but shortly after that, the height of the Cinnamon menu gets reduced (in order to fit under the virtual keyboard). Opening the virtual keyboard causes all icons (eg. mounted drives) to drop below the virtual keyboard. After closing the virtual keyboard, they move back to the original position. Why is the virtual keyboard popping all the open menu's away?

**Comment by clefebvre** [comment]
Upon investigation...

When the virtual keyboard pops up, the work area of the screen is reduced. The PopupResizeHandler reacts to motion event, so nothing happens until the menu is hovered. When that happens, the handler calls the _onBoxResized callback in the menu applet.. which saves the new menu height into the JSON preferences,  so these become permanent.

I tried to modify the callback to include a boolean in the arguments, to make this callback call not save to JSON in the case of a motion event. That worked but it created other issues.

Another approach I took was simply to not resize the menu when the workarea was smaller IF the virtual keyboard was visible. This looks like a good compromise to me.

This fix is available for testing at https://github.com/linuxmint/cinnamon/commit/6e33b8757f21c673e5a63a80f2f4e678e67885fe.
